### PR TITLE
Set correct npm version, use latest vscode extension versions

### LINKED ISSUE #449: Local build fails due to conflicting peer dependencies
With npm version `8.19.4`, I get the following errors when running `npm install`:

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: gatsby-remark-autolink-headers@4.0.0
npm ERR! Found: gatsby@2.23.3
npm ERR! node_modules/gatsby
npm ERR!   gatsby@"^2.13.3" from the root project
npm ERR!   peer gatsby@"^2.0.0" from babel-plugin-remove-graphql-queries@2.9.5
npm ERR!   node_modules/babel-plugin-remove-graphql-queries
npm ERR!     babel-plugin-remove-graphql-queries@"^2.9.5" from gatsby@2.23.3
npm ERR!     babel-plugin-remove-graphql-queries@"^2.9.5" from gatsby-plugin-typescript@2.4.6
npm ERR!     node_modules/gatsby-plugin-typescript
npm ERR!       gatsby-plugin-typescript@"^2.4.6" from gatsby@2.23.3
npm ERR!       1 more (gatsby-admin)
npm ERR!   9 more (gatsby-admin, gatsby-interface, gatsby-plugin-emotion, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer gatsby@"^3.0.0-next.0" from gatsby-remark-autolink-headers@4.0.0
npm ERR! node_modules/gatsby-remark-autolink-headers
npm ERR!   gatsby-remark-autolink-headers@"^4.0.0" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: gatsby@3.15.0
npm ERR! node_modules/gatsby
npm ERR!   peer gatsby@"^3.0.0-next.0" from gatsby-remark-autolink-headers@4.0.0
npm ERR!   node_modules/gatsby-remark-autolink-headers
npm ERR!     gatsby-remark-autolink-headers@"^4.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Looks like we have conflicting peer dependencies. The build works, if I ignore peer dependency issues by running `npm install --legacy-peer-deps` instead.

### PR DESCRIPTION
**Pull request by danieldietrich**
Avoid #449 when using Gitpod by setting node version v14 aka lts/fermium (currently npm version 6.14.18)

@planger however ... I understand that it does not really fix the original issue of supporting newer node/npm versions. this is just a quick fix to be able to contribute to the current site

### CODE REVIEW

**Review by planger — APPROVED** [review]
Thanks @danieldietrich! This change looks good to me and I'm happy to approve!

However, I think we should keep https://github.com/eclipse-theia/theia-website/issues/449 to track the task of supporting more current node versions.
Also, I think we should mention in the [readme](https://github.com/eclipse-theia/theia-website/blob/master/README.md?plain=1#L13) that building locally requires node 14.x.

Thank you!

**Comment by danieldietrich** [comment]
Thank you for the quick reply @planger!

> (...) we should keep #449 to track the task of supporting more current node versions.

That makes absolutely sense to me!

> (...) we should mention in the readme that building locally requires node 14.x

Yep, I've missed that. I've added another commit.
Please let me know if something needs to be changed or rephrased.

Thank you Philip & have a nice weekend

File: README.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ The easiest is to use Gitpod, which has everything readily setup to start workin
 
 ### Work locally
 
+Building locally requires node 14.x.
```
Comment (planger) [inline README.md:15]:
```suggestion
Building locally requires node 14.x. Alternatively, you can use `npm i --legacy-peer-deps` to ignore conflicting peer dependencies.
```

**Review by planger — APPROVED** [review]
Thank you very much for the fast turnaround! I just made a suggestion to mention that ignoring peer dependency conflicts works too.

I think it would make sense to commit the modified `package-lock.json`. Thank you!

**Pull request merged**